Modules/IO/DCMTK/src/itkDCMTKImageIO.cxx

```diff
@@ -170,30 +170,30 @@ readNoPreambleDicom( std::ifstream & file )  // NOTE: This file is duplicated in
     long length=std::numeric_limits<long>::max();
     const std::string vr{vrcode};
     if ( vr == "AE" || vr == "AS" || vr == "AT"
-         || vr == "CS" || vr == "DA" || vr == "DS"
-         || vr == "DT" || vr == "FL" || vr == "FD"
-         || vr == "IS" || vr == "LO" || vr == "PN"
-         || vr == "SH" || vr == "SL" || vr == "SS"
-         || vr == "ST" || vr == "TM" || vr == "UI"
-         || vr == "UL" || vr == "US"
+      || vr == "CS" || vr == "DA" || vr == "DS"
+      || vr == "DT" || vr == "FL" || vr == "FD"
+      || vr == "IS" || vr == "LO" || vr == "PN"
+      || vr == "SH" || vr == "SL" || vr == "SS"
+      || vr == "ST" || vr == "TM" || vr == "UI"
+      || vr == "UL" || vr == "US"
         )
     {
+      // Explicit VR (value representation stored in the file)
       unsigned short uslength=0;
       file.read(reinterpret_cast<char *>(&uslength),sizeof(unsigned short));
       ByteSwapper<unsigned short>::SwapFromSystemToLittleEndian(&uslength);
       length = uslength;
     }
-    else {
-      //Read the reserved byte
-      unsigned short uslength=0;
-      file.read(reinterpret_cast<char *>(&uslength),sizeof(unsigned short));
-      ByteSwapper<unsigned short>::SwapFromSystemToLittleEndian(&uslength);
+    else
+    {
+      // Implicit VR (value representation not stored in the file)
+      char lengthChars[4] = { vrcode[0], vrcode[1], '\0', '\0' };
```

Comment:
Will this trick work on a big endian machine? I think big/little endian compatibility was the point of the old code.

Reply:
I see no reason why it wouldn't. There is a ByteSwapper call for this as well. I tried finding a big endian file among those that failed to load, but all that test data is little-endian.

Reply:
I used DCMTK to convert it to big endian and it works.

Reply:
I thought that the removed `ByteSwapper` call was for this. Now I see there is a `ByteSwapper` call for this on line 194.